includes/gateway/class-omise-payment-billpayment-tesco.php

```diff
@@ -0,0 +1,226 @@
+				<p><?php echo __( 'Use this barcode to pay at Tesco Lotus.', 'omise' ); ?></p>
+				<div class="omise-billpayment-tesco-barcode">
+					<?php
+					if ( 'email' === $context ) {
+						$barcode_html = $this->barcode_svg_to_html( $barcode_svg );
+						echo $barcode_html;
+					} else {
+						echo $barcode_svg;
+					}
+					?>
+				</div>
+				<small class="omise-billpayment-tesco-reference-number">
+					<?php
+					echo sprintf(
+						'| &nbsp; %1$s &nbsp; 00 &nbsp; %2$s &nbsp; %3$s &nbsp; %4$s',
+						$charge['source']['references']['omise_tax_id'],
+						$charge['source']['references']['reference_number_1'],
+						$charge['source']['references']['reference_number_2'],
+						$charge['amount']
+					);
+					?>
+				</small>
+			</div>
+			<?php
+		}
+
+		/**
+		 * @param WC_Order $order
+		 *
+		 * @see   woocommerce/templates/emails/email-order-details.php
+		 * @see   woocommerce/templates/emails/plain/email-order-details.php
+		 */
+		public function email_barcode( $order ) {
+			$this->display_barcode( $order, 'email' );
+		}
+
+		/**
+		 * Convert a given SVG Bill Payment Tesco's barcode to HTML format.
+		 *
+		 * Note that the SVG barcode contains with the following structure:
+		 *
+		 * <?xml version="1.0" encoding="UTF-8"?>
+		 * <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="515px" height="90px" viewBox="0 0 515 90" version="1.1" preserveAspectRatio="none">
+		 *   <title>** reference number **</title>
+		 *   <g id="canvas">
+		 *     <rect x="0" y="0" width="515px" height="90px" fill="#fff" />
+		 *     <g id="barcode" fill="#000">
+		 *       <rect x="20" y="20" width="2px" height="50px" />
+		 *       ... (repeat <rect> node for displaying barcode) ...
+		 *       <rect x="493" y="20" width="2px" height="50px" />
+		 *     </g>
+		 *   </g>
+		 * </svg>
+		 *
+		 * The following code in this method is to read all <rect> nodes' attributes under the <g id="barcode"></g>
+		 * in order to replicate the barcode in HTML <div></div> element.
+		 *
+		 * @param  string $barcode_svg
+		 *
+		 * @return string  of a generated Bill Payment Tesco's barcode in HTML format.
+	     */
+		public function barcode_svg_to_html( $barcode_svg ) {
+			$xml       = new SimpleXMLElement( $barcode_svg );
+			$xhtml     = new DOMDocument();
+			$prevX     = 0;
+			$prevWidth = 0;
+
+			// Read data from all <rect> nodes.
+			foreach ( $xml->g->g->children() as $rect ) {
+				$attributes = $rect->attributes();
+				$width      = $attributes['width'];
+				$margin     = ( $attributes['x'] - $prevX - $prevWidth ) . 'px';
+
+				// Set HTML attributes based on <rect> node's attributes.
+				$divRect = $xhtml->createElement( 'div' );
+				$divRect->setAttribute( 'style', "float:left; position:relative; height:50px; width:$width; background-color:#000; margin-left:$margin" );
```

Comment:
If this is printed with 'backgrounds' switched off (as is sometimes a common option on printer settings) - will the barcode still show?

Reply:
The "background switched off" printing mode will cancel the css `background`?
If so then no, the barcode will just show empty white space.

Any suggestion?

Reply:
Maybe make the 'black block' `div`s have a solid black border on one vertical side that is of the required width instead of relying on the background?

Reply:
Not 100% sure how the 'remove background' feature works, but it's generally to conserve ink/toner by only printing content as opposed to blocks of background colour/image - so I suspect it may well hide css `background`s

Reply:
@jonrandy  The "border width" approach is pretty interesting.
Let me test a bit more on this.

Thanks for the suggestion 👍

Reply:
Adding
```
box-sizing: border-box
width: 2px
border-left: 2px solid #000
```
to a `div` gets you a solid div 2px wide without using a background colour

Reply:
@jonrandy 👍 cool!
I didn't go with `box-sizing`. Last time I checked, many email clients don't support this feature, except Gmail. But done. Can you have a look again?

Thanks :D